src/components/Alert/Alert.d.ts

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+
+declare type AlertType =
+| "primary"
+| "secondary"
+| "success"
+| "info"
+| "warning"
+| "danger";
+
+declare namespace Alert {
+    export interface AlertProps extends React.Props<Alert> {
+      className?: string;
+      type: AlertType;
+      icon?: string;
+      hasExtraSpace?: boolean;
+      isDismissible?: boolean;
+      avatar?: string;
+    }
+}
+declare class Alert extends React.Component<Alert.AlertProps> { }
```

Comment:
This is also not too big but since most of the components are stateless you could consider using `React.StatelessComponent` or `React.SFC` but this would mean using `const/var` instead of `class` making line 12 to throw an error.